16 api retrive detail inn

buihieu028 opened the pull request:
Xử lý API detail Inn và sửa lại bảng InnTable

GuchiGangz committed:
add/api - find inn by id

Bổ sung api  tìm NHÀ TRỌ bằng id

GuchiGangz committed:
update/inn module

Bổ sung thêm 4 trường open_hour+minute và close_hour+minute cho bảng InnTable
Sửa lại seed factory cho bảng inn

DinnieJ commented on api/database/migrations/2021_02_23_074519_update_inn_table_open_close_hour.php:
```diff
@@ -0,0 +1,37 @@
+    {
+        //
+        Schema::table('tb_inn', function (Blueprint $table) {
+            $table->integer('open_hour')->nullable(true)->after('electric_price');
```
nullable default là true nhé, k phải đặt thế này đâu :))

DinnieJ commented on api/app/Http/Controllers/Inn/InnController.php:
```diff
@@ -0,0 +1,40 @@
+        if (!$inn) {
+            return response()->json(null, 404);
+        }
+        return response()->json($inn, 200);
```
mình cần lấy cả những căn phòng của nhà trọ này nữa
Thế này mới chỉ có lấy thông tin nhà  trọ only

GuchiGangz committed:
update/api InnDetail

Bổ sung thêm trường room của mỗi inn tại response trả về

GuchiGangz committed:
Update 2021_02_23_074519_update_inn_table_open_close_hour.php

Sửa lại nullable thành default value của nó

GuchiGangz committed:
add/RoomTableSeeder

Bổ sung các lệnh để fake data cho bảng RoomTable

DinnieJ commented on api/app/Http/Resources/InnDetailResource.php:
```diff
@@ -0,0 +1,35 @@
+            'location' => $this->location,
+            'status' => $this->status,
+            'rooms' => $this->rooms,
+            'created_at' => $this->created_at->format('d . m . Y'),
```
fix format dd-mm-yyyy đi

buihieu028 replied:
just d-m-Y

DinnieJ commented on api/database/seeds/RoomTableSeeder.php:
```diff
@@ -0,0 +1,17 @@
+     */
+    public function run()
+    {
+        //
```
Không thấy InnTableSeeder đâu, viết code factory cho inn luôn vào đây

DinnieJ commented on api/app/Http/Controllers/Inn/InnController.php:
```diff
@@ -0,0 +1,44 @@
+class InnController extends BaseController
+{
+    //
+    protected $innRepository;
```
Implement chỉ cần interface thôi, k cần cả repo kia

buihieu028 replied:
Fixed

GuchiGangz committed:
Update/Inn module

Sửa lại format date cho 3 trường created_at , updated_at và verified_at

Sửa lại controller , chỉ cần impl repository

DinnieJ commented on api/app/Models/Room.php:
```diff
@@ -8,10 +8,17 @@ class Room extends Model
+
     protected $fillable = [
-        'title', 'inn_id', 'room_type_id', 'price', 'acreage', 'description', 'verified', 'verified_at', 'available', 'status'
+        'title', 'inn_id', 'room_type_id', 'price', 'acreage', 'description', 'verified', 'verified_at', 'available', 'status', 'created_at', 'updated_at'
```
có nhất thiết phải để fillable ở đây k, casts là được rồi mà

GuchiGangz committed:
Update Room.php

Gỡ 2 trường created_at và updated_at

DinnieJ commented on api/app/Models/Room.php:
```diff
@@ -8,10 +8,17 @@ class Room extends Model
     ];
 
+
     public function inns()
```
BelongsTo chỉ có 1 thôi, sao lại để inn số nhiều :))

GuchiGangz committed:
Update Room.php

Sửa lại inns thành inn vì nhiều room thuộc một inn
